src/TextInput.jsx

```diff
@@ -16,18 +16,24 @@ class TextInput extends React.Component {
 
 	onChange(e) {
 		this.setState({ value: e.target.value });
+		this.props.onChange(e);
+	}
+
+	getValue() {
```

Comment:
no need for a 'getter' on a (pseudo) read-only public property. A parent can read `theInput.state.value` just as easily as `theInput.getValue()`.